docs/plugin-eval.md

```diff
@@ -0,0 +1,221 @@
+
+```JSON
+{
+  "plugins": [
+    "openai.calendar.1",
+  ],
+  "ideal": [
+    "You have a meeting with the CEO at 2pm."
+  ],
+  "input": [
+    {
+      "role": "plugin",
+      "namespace": "MyCalendar",
+      "endpoint": "addMeeting",
+      "content": {"time": "2pm", "title": "Meeting with CEO"},
+      "include_in_conversation": false
+    },
+    {
+      "role": "system",
+      "content": "You are a helpful assistant."
+    },
+    {
+      "role": "user",
+      "content": "List all of my meetings"
+    },
+    {
+      "role": "plugin",
+      "namespace": "MyCalendar",
+      "endpoint": "listMeetings",
+      "content": {}
+    }
+  ]
+}
+```
+
+#### Internally Populated
+
+If a plugin is updated within the same conversation as the evaluation, it has been internally populated. In this scenario, the model's behavior is evaluated based on its ability to accurately query the plugin for updated information rather than relying on its assumed state of the plugin. 
+
+For instance, after adding two items to a todolist, the model should still use the todolist plugin to check the entire day's schedule instead of relying on the conversation state.  Below, we have an example of a plugin with some externally populated state ("Call a friend"), as well as two internally populated items - getting milk and picking up the dry cleaning.
+
+The below example is a bit limited, as our local plugins code currently does not support asking the assistant which action should be next.
+
+```JSON
+{
+  "plugins": [
+    "openai.todo.1"
+  ],
+  "ideal": [
+    "The list should include getting milk, picking up the dry cleaning, and calling a friend"
+  ],
+  "input": [
+    {
+      "role": "plugin",
+      "namespace": "TodoList",
+      "endpoint": "addItem",
+      "content": {"name": "Call a friend"},
+      "include_in_conversation": false
+    },
+    {
+      "role": "system",
+      "content": "You are a helpful assistant."
+    },
+    {
+      "role": "user",
+      "content": "Can you add 'Get milk' to my todo list?"
+    },
+    {
+      "role": "plugin",
+      "namespace": "TodoList",
+      "endpoint": "addItem",
+      "content": {"name": "Get milk"},
+    },
+    {
+      "role": "user",
+      "content": "Can you add 'Pickup the dry cleaning' to my todo list?"
+    },
+    {
+      "role": "plugin",
+      "namespace": "TodoList",
+      "endpoint": "addItem",
+      "content": {"name": "Pickup the dry cleaning"},
+    },
+    {
+      "role": "user",
+      "content": "What items are on my todo list?"
+    },
+    {
+      "role": "plugin",
+      "namespace": "TodoList",
+      "endpoint": "listItems",
+      "content": {}
+    }
+  ]
+}
+```
+
+### Evaluation Workflows
+
+The interaction between the user and the model is an important consideration when evaluating the model's behavior. We've outlined a few interesting workflows below.
```

Comment:
This section is good! Makes it clear what a plugin eval entails, maybe we have this higher in the doc.
Also, it would be good to have examples of how to write any of these, and also did we discussed another evaluation workflow, which is given a conversation up until current turn (including plugin invocations), does the model do the right thing with the information presented from the plugins?